internal/k8s/configuration.go

```diff
@@ -765,14 +773,16 @@ func (c *Configuration) buildListenersAndTSConfigurations() (newListeners map[st
 // GetResources returns all configuration resources.
 func (c *Configuration) GetResources() []Resource {
 	return c.GetResourcesWithFilter(resourceFilter{
-		Ingresses:      true,
```

Comment:
There are changes related to TransportServers in the following files. Are transport servers intimately connected to the reload functionality?

Reply:
good question. this is needed for this new function https://github.com/nginxinc/kubernetes-ingress/blob/18b5aba08cffa5734ad316c560d1977dd93402cf/internal/k8s/controller.go#L640 
the function regenerates all the config, reloads NGINX and then reports events and updates statuses

if the first reload finishes with an error, that would be reflected in the statuses and events of the resources

Reply:
Understood - but why does it seem trasnport servers are a special case? The added lines all have to do with the transport server - the ingress and virtual server were there previously, but the transport server was added.

Reply:
Let me clarify that. hope it's not confusing.

The PR ensures that after all the initial config is generated, the IC reloads NGINX and emits events and updates statuses of the resources with the reload status (and any additional warnings that could be generated for resources). To accomplish that, the PR reuses the existing code that was previously in `syncConfigMap` (processes changes to the ConfigMap). That code would gather all resources,  re-generate all configs and reload NGINX  (via Configurator) and emit events and update statuses. The PR put that code in the new `updateAllConfigs()` method of the Controller. So that it can still be called from `syncConfigMap`. And we also call in from `sync()` method when the queue is drained - this is what makes that first reload.

Note that we need to re-generate the Config, because we want to report events and update statuses with warnings and errors, and some of the reported warnings are only discovered in the config-generation phase in the Configurator. (Note that UpdateConfig of Configurator returns warnings).

However, the code previously lacked reporting statuses and events for TransportServer (omission, not a bug),  so that's why I added it.

Technically, we don't need to regenerate the TransportServer configs yet in Configurator, because: https://github.com/nginxinc/kubernetes-ingress/blob/219d2479be3d38665efdf92fa2dbfc774eb01cbe/internal/configs/configurator.go#L1085:
- changes to the ConfigMap don't affect the TransportServer configs directly
- Configurator doesn't generate Warnings for TransportServers -- https://github.com/nginxinc/kubernetes-ingress/blob/219d2479be3d38665efdf92fa2dbfc774eb01cbe/internal/configs/configurator.go#L1086

I removed that code that I initially added in the PR. Note that in the future, we will need to regenerate the config. For example, we will soon support custom templates for the TransportServers in the ConfigMap, which will make it necessary. I added an explanatory comment there, because it's not obvious why we don't re-generate TransportServer configs.

Reply:
Gotcha, thanks for the explanation.